unstoppable-ios-app/domains-manager-ios/Services/PullUpViewService.swift

```diff
@@ -982,30 +984,112 @@ extension PullUpViewService: PullUpViewServiceProtocol {
                                         useCase: .pullUp)
     }
     
-    func showBadgeInfoPullUp(in viewController: UIViewController, badgeDisplayInfo: DomainProfileViewController.DomainProfileBadgeDisplayInfo) {
-        let badge = badgeDisplayInfo.badge
-        var selectionViewHeight: CGFloat = 256
-        let description = badge.description
-        let labelWidth = UIScreen.main.bounds.width - 32 // 16 * 2 side offsets
-        let descriptionHeight = description.height(withConstrainedWidth: labelWidth,
-                                                   font: .currentFont(withSize: 16,
-                                                                      weight: .regular))
-        selectionViewHeight += descriptionHeight
+    func showBadgeInfoPullUp(in viewController: UIViewController,
+                             badgeDisplayInfo: DomainProfileViewController.DomainProfileBadgeDisplayInfo,
+                             domainName: String) {
         Task {
+            let badge = badgeDisplayInfo.badge
+            var selectionViewHeight: CGFloat = 256
+            let labelWidth = UIScreen.main.bounds.width - 32 // 16 * 2 side offsets
+            
+            // Load badge icon
             var badgeIcon = badgeDisplayInfo.defaultIcon
             if let url = URL(string: badgeDisplayInfo.badge.logo),
                let image = await appContext.imageLoadingService.loadImage(from: .url(url, maxSize: Constants.downloadedIconMaxSize),
                                                                           downsampleDescription: nil) {
                 badgeIcon = image
             }
             
+            var leaderboardItems = [BadgeLeaderboardSelectionItem]()
```

Comment:
This code block seems huge. Maybe break up into several private methods?

Reply:
Updated